OSDOCS-3544 & 3545: CSI Migration of Cinder and Azure Disk to GA (#45025)
openshift/openshift-docs: modules/persistent-storage-csi-migration-enable.adoc

:_content-type: PROCEDURE
[id="persistent-storage-csi-migration-enable_{context}"]
= Manually enabling CSI automatic migration

> If you want to test Container Storage Interface (CSI) migration in development or staging {product-title} clusters, in-tree to CSI migration must be manually enabled for the following in-tree volume types:

Reviewer: Avoiding passive:
```suggestion
If you want to test Container Storage Interface (CSI) migration in development or staging {product-title} clusters, you must manually enable in-tree to CSI migration for the following in-tree volume types:
```

Author: Done

* AWS Elastic Block Storage (EBS)

* Google Compute Engine Persistent Disk (GCE-PD)

* VMware vSphere Disk

* Azure File

> CSI migration for these volume types is supported at the Technology Preview (TP) level.

Reviewer: Usually stated this way. I am also wondering if perhaps you could just omit this line entirely since it's in the big red "READ ME" box?
```suggestion
CSI migration for these volume types is supported as a Technology Preview (TP) feature.
```

Author: Removed sentence as added this statement to TP boilerplate note.

[NOTE]
====

> Enabling Container Storage Interface (CSI) automatic migration drains, and then restarts, all nodes in the cluster in sequence. This might take some time.

Reviewer 2: I would highlight the fact that this is section is only intended for testing purposes (testing CSI migration before being GA). For production system, the CSI migration will be automatically enabled and no manual operation is required (once the migration is GA). It's mentioned earlier in the doc but it would be best to avoid any confusion.

After turning on Technology Preview features using feature gates, they cannot be turned off. As a result, cluster upgrades are prevented.
====
+
The following configuration example enables CSI automatic migration for all CSI drivers supported by this feature that are currently in Technology Preview (TP) status:

> +

Author: I've added a paragraph at the beginning of the Enabling section to highlight this.

> +

Author: This is now covered in the extensive re-write that I've done.
